tests/Type/Doctrine/Query/QueryResultTypeWalkerTest.php

```diff
@@ -1233,6 +1235,12 @@ public function getTestData(): iterable
 							SQRT(1)
 				FROM		QueryResult\Entities\Many m
 			',
+			InstalledVersions::satisfies(new VersionParser(), 'doctrine/dbal', '<3.4') && PHP_VERSION_ID >= 80100
+				? 'sqrt(): Passing null to parameter #1 ($num) of type float is deprecated'
+				: null,
+			InstalledVersions::satisfies(new VersionParser(), 'doctrine/dbal', '>=3.4') && PHP_VERSION_ID >= 80100
+				? 'sqrt(): Passing null to parameter #1 ($num) of type float is deprecated'
+				: null,
```

Comment:
That's the result I got indeed.

With older versions of DBAL:
```
Doctrine\DBAL\Exception: An exception occurred while executing 'SELECT SQRT(m0_.intColumn) AS sclr_0, SQRT(NULLIF(m0_.intColumn, 1)) AS sclr_1, SQRT(1) AS sclr_2 FROM Many m0_':

sqrt(): Passing null to parameter #1 ($num) of type float is deprecated
```

This is because 
- In old versions of DBAL a catch Throwable is done
- With recent version, it's catching a specific exception

And the logic behind the 
```
convertDeprecationsToExceptions="true"
```
option is to throw a `DeprecatedException` when a deprecation occur so it's was caught by doctrine try/catch in older versions.

But I was wrong about the version constraint. This changed between dbal 2 and dbal 3.

Reply:
Yeah but why is it an error at all? Whose bug is it? Are we calling it correctly? Is there something to fix in Doctrine?

Reply:
Isn't it about sample data inserted into the database? Can we fix the inserted data?

Reply:
It's because of the `SQRT(NULLIF(m.intColumn, 1))` part which is calling `sqrt` on `null`.
But removing this part of the sql won't test anymore that PHPStan resolve correctly the type to add `|null` when
- This sql was working without any deprecation on older than 8.1 php versions
- This sql is still working with a deprecation after php 8.1 for recent dbal versions

Do you prefer to remove the SQL-null related part ?